packages/core/admin/server/services/project-settings.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const fs = require('fs');
+const { pick } = require('lodash');
+
+const PROJECT_SETTINGS_FILE_INPUTS = ['menuLogo'];
+
+const parseFilesData = async files => {
+  const formatedFilesData = {};
+
+  await Promise.all(
+    PROJECT_SETTINGS_FILE_INPUTS.map(async inputName => {
+      const file = files[inputName];
+
+      // Skip empty file inputs
+      if (!file) {
+        return;
+      }
+
+      const getStream = () => fs.createReadStream(file.path);
+
+      // Add formated data for the upload provider
+      formatedFilesData[inputName] = strapi
+        .plugin('upload')
+        .service('upload')
+        .formatFileInfo({
+          filename: file.name,
+          type: file.type,
+          size: file.size,
+        });
+
+      // Add image dimensions
+      Object.assign(
+        formatedFilesData[inputName],
+        await strapi
+          .plugin('upload')
+          .service('image-manipulation')
+          .getDimensions({ getStream })
+      );
+
+      // Add file path, and stream
+      Object.assign(formatedFilesData[inputName], {
+        stream: getStream(),
+        tmpPath: file.path,
+        provider: strapi.config.get('plugin.upload').provider,
+      });
+    })
+  );
+
+  return formatedFilesData;
+};
+
+const getProjectSettings = async () => {
+  const store = await strapi.store({ type: 'core', name: 'admin' });
+  const projectSettings = await store.get({ key: 'project-settings' });
+
+  // Filter file input fields
+  PROJECT_SETTINGS_FILE_INPUTS.forEach(inputName => {
+    if (!projectSettings[inputName]) {
+      return;
+    }
+
+    projectSettings[inputName] = pick(projectSettings[inputName], [
+      'name',
+      'url',
+      'width',
+      'height',
+      'ext',
+      'size',
+    ]);
+  });
+
+  return projectSettings;
+};
+
+const uploadFiles = async files => {
```

Comment:
(In theory since you're only calling this method from the `updateProjectSettings` which is itself only called by the admin controller that has a default value for the files, you don't really need it right now, but it's a service method that will probably be called by another service or controller which won't necessarely make the undefined check for you)

Reply:
Noted thanks ;)